components/scream/src/physics/spa/tests/spa_main_test.cpp

```diff
@@ -59,7 +59,7 @@ TEST_CASE("spa_read_data","spa")
 {
   using C = scream::physics::Constants<Real>;
   static constexpr auto P0 = C::P0;
-  static constexpr auto test_tol = C::macheps*1e1;
+  static constexpr auto test_tol = C::macheps*1e2;
```

Comment:
It looks like this tolerance is only used in places where a tolerance should not even be needed (namely, the  case where there's no time/vertical interpolation).

It's true that that's the check that is failing, but I think the tolerance should not be the issue. In fact, I believe those tests should pass with tol=0, even in finite precision arithmetic. That's because `y2=y1+dy/dx*(x-x1)==y1` if x==x1, regardless of roundings. Similarly for time interpolation. Allowing a tolerance might actually be a bad idea (covering up failing comparisons that should not fail).

Reply:
Note, for time interpolation, `t_norm = (t_now - t_beg) / t_len`, so for the case t=t_beg we clearly expect exact results. One may argue that for `t=t_end` we could get `t_norm` slightly different from 1.0. That's true for general doubles, but, at least in this test, we should have t_now, t_beg, and t_end integers, so they can be represented exactly. That is, when we do
```
ts += util::days_in_month(ts.get_year(),ts.get_month())*86400;
spa_time_state.t_now = ts.frac_of_year_in_days();
```
we should have `t_now -t_beg` be _exactly_ `t_len`. So even for the case where t equals the end of the month (or, as it's probably better to say, for the beg of the next month), we should have _exact_ time interpolation.

Reply:
It might also be beneficial to add some checks on the time state. SInce tests pass on other archs, I suspect they are ok, but better stress test the test. E.g., stuff like
```
REQUIRE (time_state.days_this_month==31); // it's January
REQUIRE (time_state.t_beg_month==0);
REQUIRE (time_state.t_now==0); // for 1st test
REQUIRE (time_state.t_now==31); // for 2nd test
```

Reply:
Maybe this is a more severe fail than I thought.  I can go ahead and delete most of this PR and just keep the `valgrind` fix in.  Which is the initialization of the `TimeState.month`.  And spend a bit more time investigating why the `==` fails.  I agree we would expect it to pass in cases where we have no time interpolation.  It's perplexing because it only fails in `release` build on `mappy`.  It passes in `release` build on all the machines I have access too.  But they are all intel compilers, so it is likely related to compiler.